### DISCUSSION TITLE
Exporting/archiving CollectionBuilder site

### DISCUSSION THREAD

**Discussion by midnitelibrary** [CollectionBuilder/collectionbuilder.github.io#121] [Q&A]
A student at University of Colorado Boulder recently defended their thesis and [part of it](https://marimccarville.github.io/winter_exhibit/) was built using CollectionBuilder.

We were wondering if anyone had advice for the best way to export/archive a copy of this site so that it can be placed into our institutional repository along with the rest of the project. Ideally, this would allow for a version of the site that could be downloaded and accessed/used without requiring any online services. I know that sites can be archived using Zenodo, but I don't know how/if this would work/lead to a working site for something built with CollectionBuilder.

Hopefully that all makes sense! I appreciate any help or advice people can provide.

**Comment by evanwill** [comment]
@midnitelibrary interesting question! 
There isn't an immediate way to output the site using Jekyll into a standalone offline version. I think there is two approaches: 

1. use wget or other site mirror tool to just capture a site mirror from the live website. That would look something like `wget --mirror --convert-links --adjust-extension --page-requisites --wait=2 --random-wait https://marimccarville.github.io/winter_exhibit/`. In theory this should work, but I tested it and it doesn't really with a cb-gh site--seems like wget's parsing to replace links isn't quite sophisticated enough, so ends up with a random broken stuff. But--seems like in theory could be figured out, and a different mirroring tool might be better.
2. use a jekyll plugin to build out a local version from the source code (this isn't possible with default Jekyll build behavior). If you have access to the source and Jekyll, there is some plugins and approaches to rewrite the links correctly so that they will work on the local file system as expected. I don't have an example off hand, maybe @dcnb remembers an option? I think it might require some internal tweaks to links in javascript, so might not be that easy. 

Before those options, I would remove the Disqus comments plug in, since that is an external dependency and won't work any longer. 

I want to think about this--it is still going to be a bit complicated for cb-gh, but seems like it would be handy to have built into cb-csv. 
We generally archive just the objects, metadata, and content in plain text, not the built out site. We capture sites in web archives rather than attempting an offline portable version.

**Reply by midnitelibrary** [reply]
@evanwill 
Thanks for the information.
As a related question, what are you using to capture sites as web archives?
(Also, Hi!)

**Reply by evanwill** [reply]
:wave: !

for web archives, we use Perma.cc mostly, we are a [registrar library](https://perma.cc/docs/libraries) which is pretty handy--let's you sort of manage your collection of archived pages. Sometimes also just do it directly in InternetArchive wayback, we have a lab account so we can keep track of what we have in there too. You can also use [wget to create standalone WARC files](https://evanwill.github.io/_drafts/notes/wget-archives.html) if you wanted to deposit that--access to them is less useful that in a big web archives.
